data_prep/project_context/context_retriever.py

```diff
@@ -231,28 +228,76 @@ def _get_dequal_type(self, fully_qualified_type: str) -> str:
 
     return dequal_type
 
+  def _get_namespace_node(self, namespace_node, current_depth: int) -> bool:
+    """Determines whether a namespace node has the target function contained within it."""
+    if namespace_node.get('name') != self._namespace_tokens[current_depth]:
+      return False
+
+    current_depth += 1
+
+    for node in namespace_node.get('inner'):
+      if current_depth == len(self._namespace_tokens):
+        if node.get('name') == self._function_name:
+          return True
+
+      if node.get('kind') == 'NamespaceDecl' and current_depth < len(
+          self._namespace_tokens):
+        if not self._get_namespace_node(node, current_depth):
+          continue
+        else:
+          return True
```

Comment:
nit: I suppose this is:
```python
        if (node.get('kind') == 'NamespaceDecl' and current_depth < len(self._namespace_tokens) and self._get_namespace_node(node, current_depth)):
          return True
```
Using extra `()` here because the conditions are too long and need linebreaks.
Autoformat **should** be able to add the linebreaks for you.